index.d.ts

```diff
@@ -0,0 +1,133 @@
+  }
+
+  export function after<L, R> (duration: number, value: string): IFuture<L, R>
+
+  export function and<L, R> (a: IFuture<L, R>, b: IFuture<L, R>): IFuture<L, R>
+
+
+  // TODO: Need Help
+  export function ap<L, R> (): IFuture<L, R>
+
+  export function attempt<L, R> (f: () => L): IFuture<L, R>
+
+  export function bimap<A, B, AA, BB> (left: (a: A) => AA,
+                                       right: (a: B) => BB): IFuture<AA, any>
+
+  export function both<A, B> (a: IFuture<A, any>, b: IFuture<B, any>): IFuture<[A, B], any>
+
+  export function cache<L, R> (f: IFuture<L, any>): IFuture<L, R>
+
+  export function chain<L, R> (fn: (s: R) => IFuture<IFuture<L, any>, any>, src: IFuture<R, any>): IFuture<L, R>
+
+  export function encase2<L, R, A, B> (fn: (a: A, b: B) => L, a: A, b: B): IFuture<L, R>
+
+  export function encase3<L, R, A, B, C> (fn: (a: A, b: B, c: C) => L, a: A, b: B, c: C): IFuture<L, R>
+
+  export function encase<L, R> (fn: FunctionReturning<L>, ...t: any[]): IFuture<L, R>
+  export function encase<L, R> (fn: FunctionReturning<L>): {(...t: any[]): IFuture<L, R>}
+
+
+  // TODO: Need Help
+  export function encaseN2<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function encaseN3<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function encaseN<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function encaseP2<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function encaseP3<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function encaseP<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function extractLeft<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function extractRight<L, R> (): IFuture<L, R>
+
+  export function fold<A, AA, B, BB> (a: (x: A) => AA, b: (x: B) => BB): IFuture<BB | AA, any>
+
+  // TODO: Need Help
+  export function fork<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function go<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function hook<L, R> (): IFuture<L, R>
+
+  export function isFuture<L, R> (fut: any): boolean
+
+  // TODO: Need Help
+  export function isNever<L, R> (): boolean
+
+  // TODO: Need Help
+  export function lastly<L, R> (): IFuture<L, R>
+
+  export function map<L, R, M> (fn: (value: L) => M, future: IFuture<L, R>): IFuture<M, R>
+
+  // TODO: Need Help
+  export function mapRej<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function never<L, R> (): IFuture<L, R>
+
+  // TODO: Need Help
+  export function node<L, R> (): IFuture<L, R>
+
+  export function of<L, R> (value: L): IFuture<L, R>
+
+  export function or<L, R> (a: IFuture<L, R>, b: IFuture<L, R>): IFuture<L, R>
+
+  // TODO: Need Help
+  export function par<L, R> (): IFuture<L, R>
+
+  export function parallel<L, R> (concurrency: number, futures: Array<IFuture<any, any>>): IFuture<Array<L>, any>
+
+  export function promise<L> (f: IFuture<L, any>): Promise<L>
```

Comment:
Promises are available globally out of the box if you are using typescript in your project.